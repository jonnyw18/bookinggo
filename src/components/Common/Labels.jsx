import React from 'react';
import PropTypes from 'prop-types';

const Labels = ({ locations }) => {
    switch( locations.placeType ) {
        case "A":
            return (
                <div className="label-container">
                    <div className="autocomplete-tag">
                        <span id="airport">Airport</span>
                    </div>
                    <div className="label-details-container">
                        <li> {locations.name} {locations.iata ? `(${ locations.iata })` : ""} </li>
                        <label
                            className="locations-label"> {locations.city ? locations.city + "," : null} {locations.country} </label>
                    </div>
                </div>
            );
        case "C":
            return (
                <div className="label-container">
                    <div className="autocomplete-tag">
                        <span id="city">City</span>
                    </div>
                    <div className="label-details-container">
                        <li> {locations.name} {locations.iata ? `(${ locations.iata})` : ""} </li>
                        <label
                            className="locations-label"> {locations.region ? locations.region + "," : null} {locations.country} </label>
                    </div>
                </div>
            );
        case "T":
            return (
                <div className="label-container">
                    <div className="autocomplete-tag">
                        <span id="station">Station</span>
                    </div>
                    <div className="label-details-container">
                        <li> {locations.name} {locations.iata ? `(${locations.iata})` : ""} </li>
                        <label
                            className="locations-label"> {locations.city}, {locations.region}, {locations.country} </label>
                    </div>
                </div>
            );
        case "D":
            return (
                <div className="label-container">
                    <div className="autocomplete-tag">
                        <span id="district">District</span>
                    </div>
                    <div className="label-details-container">
                        <li> {locations.name} {locations.iata ? `(${locations.iata})` : ""} </li>
                        <label className="locations-label"> {locations.city ? locations.city + ',' : null} {locations.region}, {locations.country} </label>
                    </div>
                </div>
            );
        default:
            return (
                <div className="label-container">
                    <div className="label-details-container">
                        <li id="no-results"> {locations.name} </li>
                    </div>
                </div>
            )
    }
};

Labels.propTypes = {
    locations: PropTypes.object.isRequired
};

export default Labels