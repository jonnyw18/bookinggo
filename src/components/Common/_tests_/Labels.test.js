import Labels from '../Labels';

describe('<Labels />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Labels
        locations={{}}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
