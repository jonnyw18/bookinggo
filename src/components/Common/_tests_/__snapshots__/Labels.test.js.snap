// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Labels /> renders correctly 1`] = `
<div
  className="label-container"
>
  <div
    className="label-details-container"
  >
    <li
      id="no-results"
    >
       
       
    </li>
  </div>
</div>
`;
